precompiles/bank/testdata/bank.go

```diff
@@ -14,12 +14,21 @@ var (
 	//go:embed BankCaller.json
 	BankCallerJSON []byte
 
+	// BankCallerHardhatContract is the compiled contract calling the Bank precompile
+	// generated by HardHat.
+	BankCallerHardhatContract evmtypes.HardhatCompiledContract
+
 	// BankCallerContract is the compiled contract of BankCaller.sol
 	BankCallerContract evmtypes.CompiledContract
 )
 
 func init() {
-	err := json.Unmarshal(BankCallerJSON, &BankCallerContract)
+	err := json.Unmarshal(BankCallerJSON, &BankCallerHardhatContract)
+	if err != nil {
+		panic(err)
+	}
+
+	BankCallerContract, err = BankCallerHardhatContract.ToCompiledContract()
```

Comment:
There are a lot of changes like this, where the wiring is adjusted to the "new" approach with the Hardhat generated JSON files, which boosts the amount of changed files 👀